com.unity.shadergraph/Editor/GraphUI/DataModel/Examples/ShaderGraphExampleTypes.cs

```diff
@@ -64,44 +53,39 @@ public static TypeHandle GetGraphType(PortHandler reader)
         public static readonly TypeHandle Texture2D = typeof(Texture2D).GenerateTypeHandle();
         public static readonly TypeHandle Texture3D = typeof(Texture3D).GenerateTypeHandle();
-        public static readonly TypeHandle DayOfWeek = typeof(DayOfWeek).GenerateTypeHandle();
         public static readonly TypeHandle GradientTypeHandle = typeof(Gradient).GenerateTypeHandle();
 
-        public static readonly Dictionary<string, TypeHandle> TypeHandlesByName = new()
+        public static readonly TypeHandle Matrix2 = TypeHandleHelpers.GenerateCustomTypeHandle("Matrix 2");
+        public static readonly TypeHandle Matrix3 = TypeHandleHelpers.GenerateCustomTypeHandle("Matrix 3");
+        public static readonly TypeHandle Matrix4 = TypeHandleHelpers.GenerateCustomTypeHandle("Matrix 4");
+
+        public static GraphType.Height GetGraphTypeHeight(TypeHandle th)
+        {
+            if (th == Matrix4) return GraphType.Height.Four;
+            if (th == Matrix3) return GraphType.Height.Three;
+            if (th == Matrix2) return GraphType.Height.Two;
+
+            return GraphType.Height.One;
+        }
+
+        public static GraphType.Length GetGraphTypeLength(TypeHandle th)
+        {
+            if (th == Matrix4 || th == TypeHandle.Vector4 || th == Color) return GraphType.Length.Four;
+            if (th == Matrix3 || th == TypeHandle.Vector3) return GraphType.Length.Three;
+            if (th == Matrix2 || th == TypeHandle.Vector2) return GraphType.Length.Two;
+
+            return GraphType.Length.One;
+        }
+
+        public static GraphType.Primitive GetGraphTypePrimitive(TypeHandle th)
         {
-            { "MissingType", TypeHandle.MissingType },
-            { "Unknown", TypeHandle.Unknown },
-            { "ExecutionFlow", TypeHandle.ExecutionFlow },
-            { "MissingPort", TypeHandle.MissingPort },
-            { "Bool", TypeHandle.Bool },
-            { "Void", TypeHandle.Void },
-            { "Char", TypeHandle.Char },
-            { "Double", TypeHandle.Double },
-            { "Float", TypeHandle.Float },
-            { "Int", TypeHandle.Int },
-            { "UInt", TypeHandle.UInt },
-            { "Long", TypeHandle.Long },
-            { "Object", TypeHandle.Object },
-            { "GameObject", TypeHandle.GameObject },
-            { "String", TypeHandle.String },
-            { "Vector2", TypeHandle.Vector2 },
-            { "Vector3", TypeHandle.Vector3 },
-            { "Vector4", TypeHandle.Vector4 },
-            { "Quaternion", TypeHandle.Quaternion },
-            { "Color", Color },
-            { "AnimationClip", AnimationClip },
-            { "Mesh", Mesh },
-            { "Texture2D", Texture2D },
-            { "Texture3D", Texture3D },
-            { "DayOfWeek", DayOfWeek },
-            { "Gradient", GradientTypeHandle }
-        };
-
-        public static IEnumerable<string> TypeHandleNames => TypeHandlesByName.Keys;
```

Comment:
None of this is actually used - it was used for old exploration stuff (`AddPortPart`, which is now entirely commented out)